umpost/conversion_driver_esm1p5.py

```diff
@@ -127,28 +128,58 @@ def convert_fields_file_list(fields_file_paths, nc_write_dir):
 
     Returns
     -------
-    None
+    succeeded: list of filepaths for successfully converted fields files
+    failed: list of tuples of form (filepath, exception) for files which failed 
+    to convert due to an allowed exception. 
     """
+    succeeded = []
+    failed = []
 
-    for fields_file_path in fields_file_paths:
+    fields_file_paths = [Path(p) for p in fields_file_paths]
 
-        fields_file_path = Path(fields_file_path)
+    for fields_file_path in fields_file_paths:
 
         nc_write_path = get_nc_write_path(fields_file_path, nc_write_dir)
 
-        print("Converting file " + fields_file_path.name)
-
         try:
             um2netcdf4.process(fields_file_path, nc_write_path, ARG_VALS)
+            succeeded.append(fields_file_path)
 
         except Exception as exc:
-            # Not ideal here - um2netcdf4 raises generic exception when missing coordinates
+            # TODO: Refactor once um2nc has specific exceptions
             if exc.args[0] in ALLOWED_UM2NC_EXCEPTION_MESSAGES.values():
-                warnings.warn("Unable to convert file: " +
-                              fields_file_path.name)
+                failed.append((fields_file_path, exc))
             else:
+                # raise any unexpected errors
                 raise
 
+    return succeeded, failed
+
+
+def report_results(succeeded, failed):
```

Comment:
An option here is to _format_ a report in this function, returning a `str`.

Then, `convert_esm1p5_output_dir()` can return `succeeded/failed`, which can be fed into the report formatter (in the main block). Then, `print` I/O for reporting is right up in the top level main block.

This should make the testing a bit simpler, as the reporting output can be tested without needing to mock `print`.

Reply:
I've had a go at this, and split it into separate `format_successes` and `format_failures` to try and make the testing easier. Let me know if these changes look ok!